cpp/include/raft/linalg/rsvd.cuh

```diff
@@ -18,13 +18,94 @@
 
 #pragma once
 
-#include "detail/rsvd.cuh"
-
 #include <raft/core/device_mdspan.hpp>
+#include <raft/linalg/detail/rsvd.cuh>
 
 namespace raft {
 namespace linalg {
 
+/**
+ * @brief randomized singular value decomposition (RSVD)
+ * @tparam math_t the data type
+ * @tparam idx_t index type
+ * @param[in]  handle:  raft handle
+ * @param[in]  in:      input matrix in col-major format. On exit this matrix is destroyed
+ *                      [dim = n_rows * n_cols]
+ * @param[out] S:       array of singular values of input matrix.
+ *                      [dim = k]
+ * @param[out] U:       optional left singular values of input matrix. Use std::nullopt to not
+ * generate it. [dim = n_rows * k]
+ * @param[out] V:       optional right singular values of input matrix. Use std::nullopt to not
+ * generate it. [dim = k * n_cols]
+ * @param[in]  k:       Rank of the k-SVD decomposition of matrix in. Number of singular values to
+ * be computed. The rank is less than min(m,n).
+ * @param[in]  p:       Oversampling. The size of the subspace will be (k + p). (k+p) is less than
+ * min(m,n). (Recommended to be at least 2*k)
+ * @param[in]  niters:  Number of iteration of power method. (2 is recommended)
+ */
+template <typename math_t, typename idx_t>
+void randomized_svd(const raft::handle_t& handle,
+                    raft::device_matrix_view<math_t, idx_t, raft::col_major> in,
```

Comment:
This should probably be `const math_t`. Even if cusolver ends up needing to change it, we should still enforce const in the APIs and cast it away before invoking cusolver (and document it as such in the doxygen).